File: internal/plan/task.go
package plan

// Task represents a single task extracted from a plan document.
type Task struct {
	ID                 string   `json:"id"`
	Title              string   `json:"title"`
	Description        string   `json:"description"`
	AcceptanceCriteria []string `json:"acceptanceCriteria"`
	Status             string   `json:"status"`
	Attempts           int      `json:"attempts"`
}

// Task status constants
const (
	TaskStatusPending    = "pending"
	TaskStatusInProgress = "in_progress"
	TaskStatusCompleted  = "completed"
	TaskStatusFailed     = "failed"
)

File: ts-port/src/core/task.ts
/**
 * Task types
 */

export type TaskStatus = "pending" | "in_progress" | "completed" | "failed";

export interface Task {
  id: string;
  title: string;
  description: string;
  acceptanceCriteria: string[];
  status: TaskStatus;
  attempts: number;
}

export function createTask(
  id: string,
  title: string,
  description: string,
  acceptanceCriteria: string[]
): Task {
  return {
    id,
    title,
    description,
    acceptanceCriteria,
    status: "pending",
    attempts: 0,
  };
}
